src/Common/ThreadProfileEvents.cpp

```diff
@@ -300,73 +327,51 @@ bool PerfEventsCounters::processThreadLocalChanges(const std::string & needed_ev
     {
-        LOG_WARNING(&Poco::Logger::get("PerfEventsCounters"), "Not enough permissions to record perf events: "
+        LOG_WARNING(&Poco::Logger::get("PerfEvents"),
+            "Not enough permissions to record perf events: "
             "perf_event_paranoid = {} and CAP_SYS_ADMIN = 0",
             perf_event_paranoid);
         return false;
     }
 
-    // check file descriptors limit
-    rlimit64 limits{};
-    if (getrlimit64(RLIMIT_NOFILE, &limits))
-    {
-        LOG_WARNING(&Poco::Logger::get("PerfEventsCounters"), "Unable to get rlimit: {} ({})", strerror(errno),
-                    errno);
-        return false;
-    }
-    UInt64 maximum_open_descriptors = limits.rlim_cur;
-
-    const size_t opened_descriptors = std::distance(
-        std::filesystem::directory_iterator("/proc/self/fd"),
-        std::filesystem::directory_iterator());
-
-    UInt64 fd_count_afterwards = opened_descriptors + events_to_open.size();
-    UInt64 threshold = static_cast<UInt64>(maximum_open_descriptors * FILE_DESCRIPTORS_THRESHOLD);
-    if (fd_count_afterwards > threshold)
-    {
-        LOG_WARNING(&Poco::Logger::get("PerfEventsCounters"), "Can't measure perf events as the result number of file descriptors ({}) is more than the current threshold ({} = {} * {})",
-            fd_count_afterwards, threshold, maximum_open_descriptors,
-            FILE_DESCRIPTORS_THRESHOLD);
-        return false;
-    }
-
-    // open descriptors for new events
+    // Open descriptors for new events.
+    // Theoretically, we can run out of file descriptors. Threads go up to 10k,
+    // and there might be a dozen perf events per thread, so we're looking at
+    // 100k open files. In practice, this is not likely -- perf events are
+    // mostly used in performance tests or other kinds of testing, and the
+    // number of threads stays below hundred.
+    // We used to check the number of open files by enumerating /proc/self/fd,
+    // but listing all open files before opening more files is obviously
+    // quadratic, and quadraticity never ends well.
     for (size_t i : events_to_open)
     {
         const PerfEventInfo & event_info = raw_events_info[i];
         int & fd = thread_events_descriptors_holder.descriptors[i];
         // disable by default to add as little extra time as possible
         fd = openPerfEventDisabled(perf_event_paranoid, has_cap_sys_admin, event_info.event_type, event_info.event_config);
 
-        if (fd == -1)
+        if (fd == -1 && errno != ENOENT)
         {
-            LOG_WARNING(&Poco::Logger::get("PerfEventsCounters"), "Perf event is unsupported: {}"
-                " (event_type={}, event_config={})",
-                event_info.settings_name, event_info.event_type,
-                event_info.event_config);
+            // ENOENT means that the event is not supported, so we don't log it
+            // for each thread. Other codes might signify an error.
```

Comment:
A bit unclear what means "we don't log it for each thread" as now there is no logic that prevents it from logging each time on each thread an event can't be opened